drivers/staging/vc04_services/vc-sm-cma/vc_sm.c

```diff
@@ -757,16 +748,17 @@ static int bcm2835_vc_sm_cma_remove(struct platform_device *pdev)
 	pr_debug("[%s]: start\n", __func__);
 	if (sm_inited) {
 		/* Remove all proc entries. */
-		//debugfs_remove_recursive(sm_state->dir_root);
+		debugfs_remove_recursive(sm_state->dir_root);
 
 		/* Stop the videocore shared memory service. */
 		vc_sm_cma_vchi_stop(&sm_state->sm_handle);
+	}
 
+	if (sm_state) {
```

Comment:
Isn't this guaranteed to be non-null?

Reply:
Hold on, might have misread the diff.

Reply:
No, I think you're right, but I'm now rereading it again.

Reply:
You're right.
Should the kzalloc in probe fail then we'll bail out and remove will never be called. There should therefore never be a path where remove gets called with sm_state non-NULL.